tests/unit/cmd/scan/test_path.py

```diff
@@ -368,6 +368,26 @@ def test_scan_path_should_detect_non_git_files(self, cli_fs_runner):
         ), "not_committed files not should have been ignored"
         assert result.exception is None
 
+    @my_vcr.use_cassette()
+    def test_scan_path_use_gitignore(self, cli_fs_runner: CliRunner) -> None:
+        """
+        GIVEN a directory containing a gitignore
+        WHEN executing a scan with --use-gitignore option
+        THEN ggshield should not scan the ignored files
+        """
+        os.makedirs("repo_git_ignore")
+        Path("repo_git_ignore/secret.js").write_text(VALID_SECRET_PATCH)
+        Path("repo_git_ignore/.gitignore").write_text("secret.js")
+        os.system("git init")
+        os.system("git add .")
+        os.system("git commit -m 'initial commit'")
```

Comment:
This is going to create a repository with what is in the current directory, so the commit is likely to be slow. Better use the `tmp_path` fixture to get a temporary directory to work in. Furthermore, instead of using `os.system()` to run git commands, you can use the `Repository` helper class.

You can find an example of this in `test_scan_context_repository()` in this file.